packages/dataviews/src/dataviews-layouts/table/column-header-menu.tsx

```diff
@@ -103,6 +103,9 @@ const _HeaderMenu = forwardRef( function HeaderMenu< Item >(
 						className="dataviews-view-table-header-button"
 						ref={ ref }
 						variant="tertiary"
+						showTooltip={ !! field.description }
+						tooltipPosition="top"
+						label={ field.description }
```

Comment:
I think `description` does what we want here.

```suggestion
						description={ field.description }
```

This means that the button will use `aria-description` instead of `aria-label`. `aria-label` is overriding the label on the button, but that isn't really the behaviour we want. `aria-description` is for further explanation of what the control is, which seems more appropriate.

I don't have NVDA, but when I test with VoiceOver, focusing the column header specifically it will read out the label first, pause, then read the description.

When I use table navigation into a cell, it just announces the column heading as "satellite". Without the fullsome description. I don't know if this is something configurable in the screen reader, but the semantics seem more correct, so theoretically it would be.

Reply:
Using the `description` prop would certainly make it accessible, but it doesn't seem to display a tooltip, so users who aren't using a screen reader won't notice the description text.

Here's one way to improve accessibility while still supporting tooltips, although the code may be a bit verbose. There may be a better way to write it.

<details><summary>Details</summary>

```javascript
<Menu.TriggerButton
	render={
		!! field.description ? (
			<Tooltip text={ field.description }>
				<Button
					size="compact"
					className="dataviews-view-table-header-button"
					ref={ ref }
					variant="tertiary"
					showTooltip
					tooltipPosition="top"
				>
					{ header }
					{ view.sort && isSorted && (
						<span aria-hidden="true">
							{ sortArrows[ view.sort.direction ] }
						</span>
					) }
				</Button>
			</Tooltip>
		) : (
			<Button
				size="compact"
				className="dataviews-view-table-header-button"
				ref={ ref }
				variant="tertiary"
			>
				{ header }
				{ view.sort && isSorted && (
					<span aria-hidden="true">
						{ sortArrows[ view.sort.direction ] }
					</span>
				) }
			</Button>
		)
	}
/>
```

</details>

Reply:
Thanks all for testing and helping! I somehow found out that we can use `shortcut` property and it should address all the concerns... could you help test it again? 🙇

Reply:
You'll also need a `description` prop to be properly read by a screen reader, so something like this:

```jsx
<Button
	size="compact"
	className="dataviews-view-table-header-button"
	ref={ ref }
	variant="tertiary"
	showTooltip={ !! field.description }
	tooltipPosition="top"
	shortcut={ field.description }
	description={ field.description }
/>
```
However, using the `shortcut` prop for something other than its intended purpose seems a bit hacky.

cc @WordPress/gutenberg-components We're trying to achieve the following with the `Button` component:

- The button has a visible text.
- When the button is focused or hovered, we want to display an additional description that is different from the button text as a tooltip.
- Screen readers need to read both the button text and the additional description.

Reply:
> However, using the shortcut prop for something other than its intended purpose seems a bit hacky.

Agree. I resorted to your solution (wrapping with `<Tooltip>`), although the code is a bit more verbose. But I think it's a more correct way.